mem0/vector_stores/azure_ai_search.py

```diff
@@ -37,42 +40,73 @@ class OutputData(BaseModel):
+            compression_type (str): Specifies the type of quantization to use.
+                Allowed values are "none", "scalar", or "binary".
+            use_float16 (bool): Whether to store vectors in half precision (Edm.Half) or full precision (Edm.Single).
         """
         self.index_name = collection_name
         self.collection_name = collection_name
         self.embedding_model_dims = embedding_model_dims
-        self.use_compression = use_compression
+        self.compression_type = compression_type.lower()
+        self.use_float16 = use_float16
+
         self.search_client = SearchClient(
             endpoint=f"https://{service_name}.search.windows.net",
             index_name=self.index_name,
             credential=AzureKeyCredential(api_key),
         )
         self.index_client = SearchIndexClient(
-            endpoint=f"https://{service_name}.search.windows.net", credential=AzureKeyCredential(api_key)
+            endpoint=f"https://{service_name}.search.windows.net",
+            credential=AzureKeyCredential(api_key),
         )
+
+        # Inject custom UserAgent header ("mem0") for tracking indexes created via mem0.
+        try:
+            self.search_client._client._config.user_agent_policy.add_user_agent("mem0")
+            self.index_client._client._config.user_agent_policy.add_user_agent("mem0")
+        except Exception as e:
+            logger.warning(f"Failed to add custom UserAgent header: {e}")
+
         self.create_col()  # create the collection / index
 
     def create_col(self):
-        """Create a new index in Azure Cognitive Search."""
-        vector_dimensions = self.embedding_model_dims  # Set this to the number of dimensions in your vector
-
-        if self.use_compression:
+        """Create a new index in Azure AI Search."""
+        # Determine vector type based on use_float16 setting.
+        if self.use_float16:
             vector_type = "Collection(Edm.Half)"
-            compression_name = "myCompression"
-            compression_configurations = [ScalarQuantizationCompression(compression_name=compression_name)]
         else:
             vector_type = "Collection(Edm.Single)"
-            compression_name = None
-            compression_configurations = []
+
+        # Configure compression settings based on the specified compression_type.
+        compression_configurations = []
+        compression_name = None
+        if self.compression_type == "scalar":
+            compression_name = "myCompression"
+            compression_configurations = [
+                ScalarQuantizationCompression(compression_name=compression_name)
+            ]
+        elif self.compression_type == "binary":
+            compression_name = "myCompression"
+            compression_configurations = [
+                BinaryQuantizationCompression(compression_name=compression_name)
+            ]
 
         fields = [
```

Comment:
No way to get the original text and not just the vector so we can store it in a searchable field? Would be nice to get the search result quality boost from using hybrid.